services/src/main/java/org/keycloak/services/clientpolicy/context/ClientCRUDContext.java

```diff
@@ -48,19 +51,39 @@ default ClientModel getClientToBeUpdated() {
      *
      * @return {@link UserModel}
      */
-    UserModel getAuthenticatedUser();
+    default UserModel getAuthenticatedUser() {
+        return null;
+    }
+
     /**
      * returns {@link UserModel} of the authenticated client.
      *
      * @return {@link UserModel}
      */
-    ClientModel getAuthenticatedClient();
+    default ClientModel getAuthenticatedClient() {
+        return null;
+    }
+
+    /**
+     * @return the newly registered client {@link ClientModel}
+     */
+    default ClientModel getRegisteredClient() {
```

Comment:
Not sure if methods "getRegisteredClient()" and "getUpdatedClient()" are necessary? Maybe it is sufficient to have a single method "getTargetClient()" ? It seems the only issue is, that this method will be null for the REGISTER event, which happens before client registration. But this can be possibly covered in the javadoc of the REGISTER event?

Reply:
@mposolda As you pointed out, we need not provide these 2 method. It is sufficient to provide `getTargetClient()` to satisfy their purposes of each implementing concrete context classes. Also, on `REGISTER` event, `getTargetClient()` returns null. Therefore, I will add comments about it onto the header of `getTargetClient()`.

- AbstractAdminClientCRUDContext
  getTargetClient() = null
  getRegisteredClient() = null
  getUpdatedClient() = null
- AdminClientRegisterContext
   getTargetClient() = null
   getRegisteredClient() = null
   getUpdatedClient() = null
- AdminClientRegisteredContext
   getTargetClient() = null
   getRegisteredClient() = registeredClient
   getUpdatedClient() = null
- AdminClientUnregisterContext
   getTargetClient() = targetClient
   getRegisteredClient() = null
   getUpdatedClient() = null
- AdminClientUpdateContext
   getTargetClient() = targetClient
   getRegisteredClient() = null
   getUpdatedClient() = null
- AdminClientUpdatedContext
   getTargetClient() = null
   getRegisteredClient() = null
   getUpdatedClient() = targetClient
-  AdminClientUpdatedContext
   getTargetClient() = targetClient
   getRegisteredClient() = null
   getUpdatedClient() = null
- AbstractDynamicClientCRUDContext
  getTargetClient() = null
  getRegisteredClient() = null
  getUpdatedClient() = null
- DynamicClientRegisterContext
   getTargetClient() = null
   getRegisteredClient() = null
   getUpdatedClient() = null
- DynamicClientRegisteredContext
   getTargetClient() = null
   getRegisteredClient() = registeredClient
   getUpdatedClient() = null
- DynamicClientUnregisterContext
   getTargetClient() = targetClient
   getRegisteredClient() = null
   getUpdatedClient() = null
- DynamicClientUpdateContext
   getTargetClient() = targetClient
   getRegisteredClient() = null
   getUpdatedClient() = null
- DynamicClientUpdatedContext
   getTargetClient() = null
   getRegisteredClient() = null
   getUpdatedClient() = targetClient
- DynamicClientViewContext
   getTargetClient() = targetClient
   getRegisteredClient() = null
   getUpdatedClient() = null